toolkit/primitives/domain/src/lib.rs

```diff
@@ -298,15 +311,37 @@ impl MainchainKeyHash {
 	pub fn from_vkey(vkey: [u8; 32]) -> Self {
 		Self(blake2b(&vkey))
 	}
+
+	pub fn from_bech32_address(address: &str) -> Result<Self, String> {
+		let (_hrp, u5data, _variant) = bech32::decode(address).map_err(|e| e.to_string())?;
+
+		let data: Vec<u8> = bech32::FromBase32::from_base32(&u5data).unwrap();
+		let hash_bytes: [u8; MAINCHAIN_KEY_HASH_LEN] =
+			data[1..1 + MAINCHAIN_KEY_HASH_LEN].try_into().unwrap();
+
+		Ok(Self(hash_bytes))
+	}
 }
 
-/// ECDSA signature of MainchainPrivateKey, 64 bytes.
+/// EDDSA signature of MainchainPrivateKey, 64 bytes.
```

Comment:
Just an EDDSA signature. Nowhere in the code it is obtained using MainchainPrivateKey